MerlinAU.sh

```diff
@@ -4797,6 +4811,96 @@ _ShowNodesMenu_()
    printf "${SEPstr}"
 }
 
+_ShowNodesMenuOptions_()
+{
+    while true
+    do
+        _ShowNodesMenu_
+        printf "\nEnter selection:  "
+        read -r nodesChoice
+        echo
+        case $nodesChoice in
+            e|exit) break
+               ;;
+            *) _InvalidMenuSelection_
+               ;;
+        esac
+    done
+}
+
+##---------------------------------------##
+## Added by ExtremeFiretop [2024-May-02] ##
+##---------------------------------------##
+_DownloadChangelogs_()
+{
+    changeLogTag="$(echo "$(nvram get buildno)" | grep -qE "^386[.]" && echo "386" || echo "NG")"
+    if [ "$changeLogTag" = "386" ]
+    then
+        wget -O "$FW_BIN_DIR/Changelog-${changeLogTag}.txt" "${CL_URL_386}"
+    elif [ "$changeLogTag" = "NG" ]
+    then
+        wget -O "$FW_BIN_DIR/Changelog-${changeLogTag}.txt" "${CL_URL_NG}"
+    fi
+    changeLogFile="${FW_BIN_DIR}/Changelog-${changeLogTag}.txt"
+    if [ ! -f "$changeLogFile" ]
+    then
+        Say "Change-log file [$changeLogFile] does NOT exist."
+        _DoCleanUp_ 1 "$keepZIPfile" "$keepWfile"
```

Comment:
This same "`_DoCleanUp_`" line is duplicated & executed later in line #4856.

Reply:
Removed as requested